src/libYARP_dev/src/ControlModeImpl.cpp

```diff
@@ -104,5 +104,12 @@ bool ImplementControlMode::setControlModes(const int n_joint, const int *joints,
 
 bool ImplementControlMode::setControlModes(int *modes)
 {
-    return raw->setControlModesRaw(modes);
+    int *modes_tmp=new int [nj];
```

Comment:
I would use `std::array` instead of dynamic allocation, here and in the other places.
See http://www.modernescpp.com/index.php/std-array-dynamic-memory-no-thanks

Reply:
Actually it is not possible to declare:
```c++
std::array<int, nj> modes_tmp;
```
because `nj` is not `constexpr`. :cry: 

But here we can use smart pointers and get rid of all `delete` :wink:
```
std::unique_ptr<int[]> modes_tmp(new int [nj]);
```

Reply:
Hi @Nicogene ,
this request would fix some bugs only.

Here is not the place to change the dynamic allocations, that will be modify with a fixed size buffer manager asap, but on devel branch. If you are interested in, please see https://github.com/valegagge/yarp/tree/fixedSizeBuffersManager.

The dynamic allocations had been a temporary solotion to fix issue (https://github.com/robotology/icub-support/issues/624) in order to continue make faster the firmware response to some user-request, parallelizing the request to all motion control devices. (see (https://github.com/robotology/yarp/pull/1668)

The fixed-size buffer-manager will be use instead of dynamic allocation as explained in: https://github.com/robotology/yarp/issues/1684.

So, here I would only to fix some wrong behaviour of these functions.
The important thing now is to understand if my changes fix bugs or not.

cheers 
Vale

Reply:
Hi @valegagge ! I see, if these dynamic allocations has to die so it is not worthy to migrate to smart pointers :wink:

Reply:
yes! I'm testing fixedsizeBufferManager! :)
So we can merge this pr?

Reply:
I'd add some lines in the changelog, and I don't know if @drdanz is going to release yarp `3.0.1` soon. Shall we include it in this release or in `3.1.1` ??